src/common/telemetry.ts

```diff
@@ -146,35 +144,15 @@ export class BaseTelemetryReporter extends Disposable {
 	private userOptIn = false;
 	private errorOptIn = false;
 	private _extension: vscode.Extension<any> | undefined;
+	private readonly appenders = new Map<string, ITelemetryAppender>();
 
 	constructor(
 		private extensionId: string,
 		private extensionVersion: string,
-		private telemetryAppender: ITelemetryAppender,
 		private osShim: { release: string; platform: string; architecture: string },
+		private readonly clientFactory: (gitpodHost: string) => Promise<BaseTelemetryClient>
 	) {
 		super();
-
-		this.updateUserOptStatus();
-
-		if (vscode.env.onDidChangeTelemetryEnabled !== undefined) {
-			this._register(vscode.env.onDidChangeTelemetryEnabled(() => this.updateUserOptStatus()));
-			this._register(vscode.workspace.onDidChangeConfiguration(() => this.updateUserOptStatus()));
-		} else {
-			this._register(vscode.workspace.onDidChangeConfiguration(() => this.updateUserOptStatus()));
-		}
-	}
-
-	/**
-	 * Updates whether the user has opted in to having telemetry collected
-	 */
-	private updateUserOptStatus(): void {
-		const telemetryLevel = getTelemetryLevel();
-		this.userOptIn = telemetryLevel === TelemetryLevel.ON;
-		this.errorOptIn = telemetryLevel === TelemetryLevel.ERROR || this.userOptIn;
-		if (this.userOptIn || this.errorOptIn) {
-			this.telemetryAppender.instantiateAppender();
-		}
```

Comment:
why remove this? we should respect user choice, i always have telemetry disabled for example